### PR TITLE
test(item): migrate tests to playwright

### PR DESCRIPTION
**Pull request by sean-perkins**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
  - Some docs updates need to be made in the `ionic-docs` repo, in a separate PR. See the [contributing guide](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#modifying-documentation) for details.
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [x] Other (please describe): Test migrations

## What is the current behavior?

Item tests are written in Puppeteer. 

Issue URL:  N/A

## What is the new behavior?

Migrates tests to Playwright

- Removes flaky exception fallback test for item counter
- Introduces new option to `setIonViewport` that allows resizing the horizontal size of the viewport to match the scroll contents (for niche tests like the form test for item). 

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: core/src/components/item/test/alignment/index.html [hunk 1]
```diff
@@ -27,7 +27,7 @@
           <ion-list-header>Leading Icons</ion-list-header>
           <ion-item>
             <ion-icon slot="start" name="time"></ion-icon>
-            <ion-datetime display-format="DDDD MMMM D YYYY hh:mm:ss a" value="2019-10-01T15:43:40.394Z"></ion-datetime>
+            <ion-datetime value="2019-10-01T15:43:40.394Z"></ion-datetime>
```
Comment (liamdebeasi) [inline core/src/components/item/test/alignment/index.html:30]: Do we need to use `ion-datetime` here? Wondering if we can use something else instead to make the screenshots a bit smaller

File: core/src/utils/test/playwright/playwright-declarations.ts [hunk 2]
```diff
@@ -95,3 +95,11 @@ export interface E2EPage extends Page {
   _e2eEventsIds: number;
   _e2eEvents: Map<number, any>;
 }
+
+export interface SetIonViewportOptions {
+  /**
+   * `true` if the viewport should be scale to match the `ion-content`
```
Comment (liamdebeasi) [inline core/src/utils/test/playwright/playwright-declarations.ts:101]:
```suggestion
   * `true` if the viewport should be scaled to match the `ion-content`
```

**Review by liamdebeasi — APPROVED** [review]
Great job!

File: core/src/components/item/test/alignment/index.html (hunk 1 above)
Comment (sean-perkins) [reply core/src/components/item/test/alignment/index.html:30]: I don't believe so, I can swap those out with inputs 👍

File: core/src/components/item/test/reorder/item.e2e.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { expect } from '@playwright/test';
+import { test } from '@utils/test/playwright';
+
+test.describe('item: reorder', () => {
+  test('should not have visual regressions', async ({ page }) => {
+    await page.goto(`/src/components/item/test/reorder`);
+
+    await page.setIonViewport();
+
+    await page.click('text=Edit');
```
Comment (amandaejohnston) [inline core/src/components/item/test/reorder/item.e2e.ts:10]: ~ Might be better to use a more stable selector, like an ID.

File: core/src/components/item/test/inputs/item.e2e.ts [hunk 4]
```diff
@@ -0,0 +1,107 @@
+import { expect } from '@playwright/test';
+import type { E2EPage } from '@utils/test/playwright';
+import { test } from '@utils/test/playwright';
+
+test.describe('item: inputs', () => {
+  test('should not have visual regressions', async ({ page }) => {
```
[the remark this answers is not in the document]
Comment (sean-perkins) [reply core/src/components/item/test/inputs/item.e2e.ts:6]: Discovered this is actually a bug (most likely occurred from v5 to v6), created the bug here: https://github.com/ionic-team/ionic-framework/issues/25483

Will create a PR to fix that first.

File: core/src/components/item/test/inputs/item.e2e.ts [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
```
Comment (amandaejohnston) [inline core/src/components/item/test/inputs/item.e2e.ts:6]: At least one of the datetimes is missing entirely here:
- `core/src/components/item/test/inputs/item.e2e.ts-snapshots/item-inputs-ios-rtl-Mobile-Firefox-linux.png`
- `core/src/components/item/test/inputs/item.e2e.ts-snapshots/item-inputs-md-ltr-Mobile-Safari-linux.png`
- `core/src/components/item/test/inputs/item.e2e.ts-snapshots/item-inputs-md-rtl-Mobile-Safari-linux.png`

You may need to do `await page.waitForSelector('.datetime-ready')`?

**Review by amandaejohnston — CHANGES_REQUESTED** [review]

**Pull request merged**